core/local/local.go

```diff
@@ -67,13 +68,20 @@ func NewExecutionScheduler(runner lib.Runner, logger *logrus.Logger) (*Execution
 
 	executorConfigs := options.Execution.GetSortedConfigs()
 	executors := make([]lib.Executor, len(executorConfigs))
-	for i, sc := range executorConfigs {
+	// Only take executor which has works.
+	n := 0
+	for _, sc := range executorConfigs {
+		if !sc.HasWork(executionState.Options.ExecutionSegment) {
+			continue
+		}
 		s, err := sc.NewExecutor(executionState, logger.WithField("executor", sc.GetName()))
 		if err != nil {
 			return nil, err
 		}
-		executors[i] = s
+		executors[n] = s
+		n++
```

Comment:
You can just use `append()` so we don't have to keep track of the index here.

Reply:
See https://github.com/loadimpact/k6/pull/1307#discussion_r365119495

Reply:
See https://github.com/loadimpact/k6/pull/1307#discussion_r365122531 :wink:

Reply:
Fixed